vm/src/stdlib/zlib.rs

```diff
@@ -89,86 +113,118 @@ mod decl {
         Ok(vm.ctx.new_bytes(encoded_bytes))
     }
 
-    fn header_from_wbits(
-        wbits: OptionalArg<i8>,
-        vm: &VirtualMachine,
-    ) -> PyResult<(Option<bool>, u8)> {
+    enum InitOptions {
+        Standard {
+            header: bool,
+            #[cfg(feature = "zlib")]
+            wbits: u8,
+        },
+        #[cfg(feature = "zlib")]
+        Gzip { wbits: u8 },
+    }
+
+    impl InitOptions {
+        fn decompress(self) -> Decompress {
+            match self {
+                #[cfg(feature = "zlib")]
+                Self::Standard { header, wbits } => Decompress::new_with_window_bits(header, wbits),
+                #[cfg(not(feature = "zlib"))]
+                Self::Standard { header } => Decompress::new(header),
+                #[cfg(feature = "zlib")]
+                Self::Gzip { wbits } => Decompress::new_gzip(wbits),
+            }
+        }
+        fn compress(self, level: Compression) -> Compress {
+            match self {
+                #[cfg(feature = "zlib")]
+                Self::Standard { header, wbits } => {
+                    Compress::new_with_window_bits(level, header, wbits)
+                }
+                #[cfg(not(feature = "zlib"))]
+                Self::Standard { header } => Compress::new(level, header),
```

Comment:
if this one doesn't share condition with the others, is this needed to be placed between `feature  = "zlib"`?

Reply:
I moved it above the `feature = "zlib"` branches.